File: iossdkdemo/SDKDemo/APIControllers/AuthenticationWithSocialMediaViewController.m
//
//  AuthenticationWithSocialMediaViewController.m
//  SDKDemoApp
//
//  Created by OnePoint Global on 01/12/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "AuthenticationWithSocialMediaViewController.h"
#import <OPGSDK/OPGSDK.h>

@interface AuthenticationWithSocialMediaViewController ()

@end

@implementation AuthenticationWithSocialMediaViewController
@synthesize googleToken,facebookToken;
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    googleToken.delegate=self;
    facebookToken.delegate=self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBAction
-(IBAction)authenticateWithGoogle:(id)sender
{
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    OPGAuthenticate *obj = [sdk authenticateWithGoogle:@"" error:&error];
    if ([obj.isSuccess intValue]==1 && error==nil)
    {
        [self showAlert:@"Authentication Successful"];

    }
    else
    {
        [self showAlert:obj.statusMessage];

    }
}

-(IBAction)authenticateWithFacebook:(id)sender
{
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    OPGAuthenticate *obj = [sdk authenticateWithFacebook:@"" error:&error];
    if ([obj.isSuccess intValue]==1 && error==nil)
    {
        [self showAlert:@"Authentication Successful"];

    }
    else
    {
        [self showAlert:obj.statusMessage];

    }
}

-(void)showAlert : (NSString*)alertMessage
{
    UIAlertController * alert = [UIAlertController
                                     alertControllerWithTitle:@"OPGSDKv2.0"
                                     message:alertMessage
                                     preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* okButton = [UIAlertAction
                                   actionWithTitle:@"OK"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action) {
                                       //Handle your yes please button action here
                                   }];
        
        [alert addAction:okButton];
        [self presentViewController:alert animated:YES completion:nil];
}

#pragma mark - TextField delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField{
    
    [textField resignFirstResponder];
    return YES;
}

@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/AuthenticationWithSocialMediaViewController.swift
//
//  AuthenticationWithSocialMediaViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 01/12/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class AuthenticationWithSocialMediaViewController: UIViewController, UITextFieldDelegate
{
    @IBOutlet weak var txtGoogleToken: UITextField!
    @IBOutlet weak var txtFacebookToken: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        self.txtGoogleToken.delegate=self
        self.txtFacebookToken.delegate=self
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func authenticateWithGoogle(_ sender: AnyObject)
    {
        let sdk = OPGSDK()        // Creating OPGSDK instance
        var obj : OPGAuthenticate?
        do {
            obj = try sdk.authenticate(withGoogle: "")
        }
        catch{
            print("Authentication Failed")         /* @"Error Occured. Contact Support!" */
            
        }
        
        if obj!.isSuccess.int32Value==1
        {
            self.showAlert(alertMessage: "Authentication Successful")
        }
        else
        {
            self.showAlert(alertMessage: (obj?.statusMessage)!)
        }
    }

    @IBAction func authenticateWithFacebook(_ sender: AnyObject)
    {
        let sdk = OPGSDK()        // Creating OPGSDK instance
        var obj : OPGAuthenticate?
        do {
            obj = try sdk.authenticate(withFacebook: "")
        }
        catch{
            print("Authentication Failed")         /* @"Error Occured. Contact Support!" */
            
        }
        
        if obj!.isSuccess.int32Value==1
        {
            self.showAlert(alertMessage: "Authentication Successful")
        }
        else
        {
            self.showAlert(alertMessage: (obj?.statusMessage)!)
        }
    }

    func showAlert(alertMessage : String)
    {
        let alertController = UIAlertController(title: "OPGSDKv2.0", message: alertMessage, preferredStyle: .alert)
            
            let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
            alertController.addAction(defaultAction)
            present(alertController, animated: true, completion: nil)
    }
    

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        return true
    }
}
